public/app/features/alerting/unified/components/receivers/form/GrafanaReceiverForm.tsx

```diff
@@ -108,10 +109,20 @@ export const GrafanaReceiverForm: FC<Props> = ({ existing, alertManagerSourceNam
     [config, existing]
   );
 
+  // if any receivers in the contact point have a "provenance", the entire contact point should be readOnly
+  const hasProvisionedItems = existing
+    ? existing.grafana_managed_receiver_configs?.some((item) => Boolean(item.provenance))
+    : false;
+
+  const readOnly = isVanillaPrometheusAlertManagerDataSource(alertManagerSourceName) || hasProvisionedItems;
+
   if (grafanaNotifiers.result) {
     return (
       <>
+        {hasProvisionedItems && <ProvisioningAlert type={'contact point'} />}
```

Comment:
I wonder if we should create e.g. an enum for the `type` prop and then build the proper message inside the `ProvisioningAlert` component. Currently, I need to know the internals of `ProvisioningAlert` component to provide a sensible value to the `type` prop

Reply:
Great suggestion, done!